nixos/modules/system/boot/kexec.nix

```diff
@@ -19,13 +22,12 @@
               exit 0
             fi
 
-            p=$(readlink -f /nix/var/nix/profiles/system)
-            if ! [[ -d $p ]]; then
-              echo "Could not find system profile for prepare-kexec"
-              exit 1
-            fi
-            echo "Loading NixOS system via kexec."
-            exec kexec --load $p/kernel --initrd=$p/initrd --append="$(cat $p/kernel-params) init=$p/init"
+            for p in /run/next-system /run/current-system; do
+              if [[ -d $p ]]; then
+                echo "Loading NixOS system via kexec."
+                exec kexec --load $p/kernel --initrd=$p/initrd --append="$(cat $p/kernel-params) init=$p/init"
+              fi
+            done
```

Comment:
I don't think we should mess with kexec at all in this PR. It's actually got some problems as is and this is only going to make it more confusing.

Reply:
Yeh I'll drop the commit (and maybe open a separate or for it)